app/gateway-oss/2.4.x/plugin-development/custom-logic.md

```diff
@@ -11,12 +11,12 @@ chapter: 3
 
 ## Introduction
 
-A Kong plugin allows you to inject custom logic (in Lua) at several
+A {{site.ce_product_name}} plugin allows you to inject custom logic (in Lua) at several
 entry-points in the life-cycle of a request/response or a tcp stream
-connection as it is proxied by Kong. To do so, one must implement one
-or several of the methods of the `base_plugin.lua` interface. Those
-methods are to be implemented in a module namespaced under:
-`kong.plugins.<plugin_name>.handler`
+connection as it is proxied by {{site.ce_product_name}}. To do so, the file
+`kong.plugins.<plugin_name>.handler` must return a table with one or
+more functions with predetermined names. Those functions will be
+invoked by {{site.ce_product_name}} at different phases when it processes traffic.
 
```

Comment:
if the `:` is removed from those function names, then it should be mentioned that they're called with a `self` (or `plugin`?) argument.

Reply:
I have added an extra paragraph indicating that they take `self` as a first param and most of them take `config` as a second param.